src/multibase.js

```diff
@@ -72,56 +64,32 @@ function isEncoded (bufOrString) {
 
   const code = bufOrString.substring(0, 1)
   try {
-    const name = getName(code)
-    return name
+    const base = getBase(code)
+    return base.getName()
   } catch (err) {
     return false
   }
 }
 
-function validNameOrCode (nameOrCode) {
-  const err = new Error('Unsupported encoding')
-
-  if (!constants.names[nameOrCode] &&
-      !constants.codes[nameOrCode]) {
-    throw err
-  }
-}
-
 function validEncode (name, buf) {
-  let decode
-
-  switch (name) {
-    case 'base58btc': {
-      decode = bs58.decode
-      buf = buf.toString() // bs58 only operates in strings bs58 strings
-    } break
-    default: throw errNotSupported
-  }
-
-  decode(buf)
+  const base = getBase(name)
+  base.decode(buf.toString())
 }
 
-function getCode (nameOrCode) {
-  validNameOrCode(nameOrCode)
-
-  let code = nameOrCode
+function getBase (nameOrCode) {
+  let base
 
   if (constants.names[nameOrCode]) {
-    code = constants.names[nameOrCode]
+    base = constants.names[nameOrCode]
+  } else if (constants.codes[nameOrCode]) {
+    base = constants.codes[nameOrCode]
+  } else {
+    throw errNotSupported
   }
 
-  return code
-}
-
-function getName (nameOrCode) {
-  validNameOrCode(nameOrCode)
-
-  let name = nameOrCode
-
-  if (constants.codes[nameOrCode]) {
-    name = constants.codes[nameOrCode]
+  if (!base.isImplemented()) {
+    throw new Error('Base ' + nameOrCode + ' is not implemented yet')
   }
 
-  return name
+  return base
```

Comment:
getBase should be different from getCode as in getCode is the code used to identify the base and getBase is the string that is the name of the base